src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp

```diff
@@ -33,13 +36,19 @@ TEST_F(KickoffFriendlyPlayTest, test_kickoff_friendly_play)
         // TODO: Implement proper validation
         // https://github.com/UBC-Thunderbots/Software/issues/1396
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
-            {
-                yield();
-            }
+            robotReceivedBall(4, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            robotInFriendlyField(0, world_ptr, yield);
+            robotInFriendlyField(1, world_ptr, yield);
+            robotInFriendlyField(2, world_ptr, yield);
+            robotInFriendlyField(3, world_ptr, yield);
+            robotInFriendlyField(4, world_ptr, yield);
+            robotInFriendlyField(5, world_ptr, yield);
```

Comment:
Could this be replaced by `robotsInFriendlyHalf` implemented here https://github.com/UBC-Thunderbots/Software/pull/1946/files?

Reply:
Yes I believe it can, I'll draft this PR for now until we merge that PR, I rather not have duplicate validation functions

Reply:
Refactored with new validation functions from https://github.com/UBC-Thunderbots/Software/pull/1946/files